### ISSUE TITLE
MSBuild <Choose> element not supported 

### ISSUE THREAD

**Issue by jmue** [Caphyon/clang-power-tools#165]
The following section is not correctly evaluated as 'choose' seems not to be supported.

```
  <PropertyGroup>
    <MyGroup>Group1</MyGroup>
  </PropertyGroup>

  <Choose>
    <When Condition="'$(MyGroup)'=='Group1'">
      <PropertyGroup>
        <MySelector>One,Two,Three</MySelector>
      </PropertyGroup>
    </When>
    <When Condition="'$(MyGroup)'=='Group2'">
      <PropertyGroup>
        <MySelector>One,Two,Three,Four</MySelector>
      </PropertyGroup>
    </When>
  </Choose>
```

**Comment by cpp-red-lion** [comment]
Hi,

Interesting... We'll look into it.
Thanks for the test case.

I was curious, are you actually using all that `MSBuild` grammar in your projects, or just playing around :)

In the past two months we've seen some pretty interesting configs from *Clang Power Tools* early adopters... I'm still amazed when I discover a new usage pattern in their MSBuild configs...

**Comment by jmue** [comment]
Hi,

yes, these issues arrise in a (big) solution in the company im working for. We are working with a lot of custom ```*.targets```, ```*.props```and configurations. I noticed your project (slides) some time ago at cppcon, but did not had the time to track down the errors yet.

Thank you for your support.

**Comment by cpp-red-lion** [comment]
COOL.
It's nice to have _diverse_ and _large-scale_ test cases for our tool.
Keeps us on the right track :)

We're doing our best to fix every issue we encounter.

**Comment by hero101111** [comment]
Just pushed a commit that enabled support for `<Choose>` MsBuild elements.

Please give a test run to the latest github version of `clang-build.ps1` and let us know it everything runs OK.

---

This fix is currently live only for the github repository.
The latest Visual Studio extension (`2.3`) does not contain the fix. It will be included in the next release.

Until then, as a workaround, you can manually replace `clang-build.ps1` in the VS extension directory with the latest from github. To locate the extension directory search for `clang-build.ps1` in `%LOCALAPPDATA%`

Don't forget to restart Visual Studio after replacing the script file.

**Comment by jmue** [comment]
Thank you!
But this time I got
```
Fehler beim Aufrufen der Methode, da [System.Xml.XmlComment] keine Methode mit dem Namen "SetAttribute" enth„lt.In 
D:\*\AppData\Local\Microsoft\VisualStudio\15.0_00aec18c\Extensions\j3nhounh.jub\clang-build.ps1:1306 Zeichen:13
```
Seems like we are using comments within ```<Choose>```

**Comment by cpp-red-lion** [comment]
Hi,

Gabi (@hero101111) fixed this issue and we're test-driving it...
Should be published in the next version, to come out this week.
We'll let you know.

Thanks for the feedback 👍

**Comment by cpp-red-lion** [comment]
This issue was fixed in `version 2.4` released today.
Thanks for you feedback 👍
